CIP-XXXX/README.md

```diff
@@ -0,0 +1,190 @@
+* InternalError - An error occurred during execution of this API call.
+* Refused - The request was refused due to lack of access - e.g. user rejected connection to this dApp.
+
+### Initial API
+
+In order to initiate communication from webpages to a user's Cardano wallet, the wallet must provide the following javascript API to the webpage. A shared, namespaced `cardano` object must be injected into the page if it did not exist already. Each wallet implementing this standard must then create a field in this object with a name unique to each wallet containing a `wallet` object with the following methods. The API is split into two stages to maintain the user's privacy, as the user will have to consent to `cardano.{walletName}.enable()` in order for the dApp to read any information pertaining to the user's wallet with `{walletName}` corresponding to the wallet's namespaced name of its choice.
+
+#### cardano.{walletName}.enable(extensions: Extension[] = [{ cip: 30 }]): Promise\<API>
+
+Errors: `ConnectorError`
+
+This is the entrypoint to start communication with the user's wallet. The wallet should request the user's permission to connect the web page to the wallet, and if permission has been granted, the [Extendable API](#extendable-api) will be returned to the dApp to use. The wallet can choose to maintain a whitelist to not necessarily ask the user's permission every time access is requested. But this behavior is up to the wallet and should be transparent to web pages using this API.
+
+Upon start, dApps can explicitly request a list of additional functionalities they expect as a list of CIP numbers capturing those extensions. Note that it's possible for two extensions to be mutually incompatible (because they provide two conflicting features). While we may try to avoid this as much as possible while designing CIPs, it is also the responsibility of wallet providers to assess whether they can support a given combination of extensions, or not. Hence, wallets aren't expected to fail should they not recognize or not support a particular combination of extensions. As a result, dApps may fail and inform their users or may use a different, less-efficient, strategy to cope with a lack of functionality.
+
+Once a connection is established, this can be called multiple times by the dApp to request access to further extensions. User permission must be asked if new extensions are requested. If this is called without new extensions requests then no user permission is required.
+
+**Note**: In order to improve backwards compatibility an empty `extensions` field should supply CIP-30 functionality, as if a CIP-30 `extension` object was provided.
+
+#### cardano.{walletName}.isEnabled(): Promise\<Extension[]>
+
+Errors: `ConnectorError`
+
+Returns a list of extensions which have been enabled by the wallet. If no wallet access has been granted then a `ConnectorError` with  `Refused` error code should be thrown.

```

Comment:
Again from a backward compatibility viewpoint, this should be allowed to return either a `bool` or a list of enabled extensions.  a bool set to true is equivalent to `CIP-30`, and should mean "This wallet does not support this CIP".  

Even though a wallet that implements this CIP should not return a bool, its a valid response from a CIP-30 wallet and it should be clear how a dapp reacts to it.

Reply:
I agree with this also, I think I had not considered backward compatibility enough... I plan to do a more significant rewrite when/if #486 is merged.